lib/string-checker.js

```diff
@@ -136,8 +150,87 @@ StringChecker.prototype = {
 
             this._errorsFound += errors.getErrorCount();
         }
+    },
 
-        return errors;
+    /**
+     * Adds parse error to the error list.
+     *
+     * @param {Errors} errors
+     * @param {Error} parseError
+     * @private
+     */
+    _addParseError: function(errors, parseError) {
+        errors.setCurrentRule('parseError');
+        errors.add(parseError.description, parseError.lineNumber, parseError.column);
+    },
+
+    /**
+     * Creates configured JsFile instance.
+     *
+     * @param {String} filename
+     * @param {String} source
+     * @param {Object} sourceTree
+     * @private
+     */
+    _createJsFileInstance: function(filename, source, sourceTree) {
+        return new JsFile(filename, source, sourceTree, {
+            es3: this._configuration.isES3Enabled(),
+            es6: this._configuration.isESNextEnabled()
+        });
+    },
+
+    /**
+     * Checks file provided with a string.
+     * @param {String} source
+     * @param {String} [filename]
+     * @returns {{output: String, errors: Errors}}
+     */
+    fixString: function(source, filename) {
+        filename = filename || 'input';
+
+        var sourceTree;
+        var parseError;
+
+        try {
+            sourceTree = JsFile.parse(source, this._esprima, this._configuration.getEsprimaOptions());
+        } catch (e) {
+            parseError = e;
+        }
+
+        if (parseError) {
+            var parseErrors = new Errors(this._createJsFileInstance(filename, source, sourceTree), this._verbose);
+            this._addParseError(parseErrors, parseError);
+            return {output: source, errors: parseErrors};
+        } else {
+            var attempt = 0;
+            var errors;
+            var file;
+
+            do {
```

Comment:
What in this `do` actually fixes the source? I see no explicit call to any fix-related function.